### PR TITLE
Add ASCII text encoder

### PR DESCRIPTION
**Pull request by itrujnara**
Add an ASCII encoder for text data

### CODE REVIEW

File: src/stimulus/data/encoding/encoders.py [hunk 1]
```diff
@@ -299,6 +299,111 @@ def decode(self, data: torch.Tensor) -> Union[str, list[str]]:
...
+        if type(data) is not torch.Tensor:
+            raise TypeError(f"Expected input data to be a tensor, got {type(data).__name__}")
+
+        tensor_1d = 1
+        tensor_2d = 2
+
+        if data.dim() == tensor_1d:
+            return "".join([chr(int(x)) for x in data.cpu().numpy() if x != 0])
+        if data.dim() == tensor_2d:
+            return ["".join([chr(int(x)) for x in d if x != 0]) for d in data.cpu().numpy().tolist()]
```
Comment (mathysgrapotte) [inline src/stimulus/data/encoding/encoders.py:402]: I would suggest to have the conversion to cpu using tensor.detach().cpu() happen on the model side rather than on the decoding side (since putting data on the device is done on the model side currently). It is also good for debugging: 
Device transfers can hurt performance/cause bugs so it's good to have all of them "visible" at model code level.

**Review by mathysgrapotte — APPROVED** [review]
LGTM
